Lib/ldap/ldapobject.py

```diff
@@ -526,7 +526,7 @@ def compare_ext_s(self,dn,attr,value,serverctrls=None,clientctrls=None):
     except ldap.COMPARE_FALSE:
       return False
     raise ldap.PROTOCOL_ERROR(
-        'Compare operation returned wrong result: %r' % (ldap_res)
+        'Compare operation returned wrong result: %r' % repr(ldap_res)
```

Comment:
This will do the `repr` operation twice – once with `repr`, and then for the `%r`.
`print('%r' % repr((1, 2)))` gives `'(1, 2)'` – note the extra quotes.

Could you try this instead?

```suggestion
        'Compare operation returned wrong result: ' + repr(ldap_res)
```